compiler/src/iree/compiler/Preprocessing/Common/CMakeLists.txt

```diff
@@ -46,6 +49,10 @@ iree_cc_library(
     MLIRMemRefDialect
     MLIRPDLDialect
     MLIRPDLInterpDialect
+    MLIRPDLLAST
+    MLIRPDLLCodeGen
+    MLIRPDLLODS
+    MLIRPDLLParser
```

Comment:
So for the `ld.lld: error: unable to find library -lMLIRPDLLParser` error on the byo_llvm job: https://github.com/iree-org/iree/actions/runs/8806295000/job/24172166753?pr=17144#step:4:15043, I suspect we'll need to add some options to https://github.com/iree-org/iree/blob/main/build_tools/llvm/byo_llvm.sh. It doesn't look like we'll need to distribute `mlir-pdll` ourselves or work that into the build/tests with how this is set up 🤞

Reply:
@ScottTodd do you mean adding the dependencies mentioned above to [mlir_config.cmake](https://github.com/iree-org/iree/blob/26b81db6a82dd37ba865421197d26f2a0768cbd6/build_tools/llvm/mlir_config.cmake) which is used by byo_llvm.sh? Or are you thinking of something different?

Reply:
I'm not sure. Those files are where I'd start looking though. I think you should be able to reproduce the CI failure by following what https://github.com/iree-org/iree/blob/main/build_tools/cmake/build_and_test_byo_llvm.sh does

Reply:
My initial thought was that since we are not building `mlir-pdll`, we wouldn't be building any of its PDLL related dependencies. That turns out to not be the case. Looking at the azure logs revealed the other PDLL libraries were being built and installed, except for `MLIRPDLLParser`.

It turns out that inside of the [cmakelists](https://github.com/llvm/llvm-project/blob/ddb67e68474092435634dd8a8e1c166f6ecd0e1a/mlir/lib/Tools/PDLL/Parser/CMakeLists.txt#L2), `MLIRPDLLParser` is added as a dependency with `llvm_add_dependency` instead of the normal `mlir_add_dependency`. I'm not entirely sure why this was done, but it was changed when tablegen became a dependency. Since tablegen isn't included in the single shared library during certain builds, neither could `MLIRPDLLParser`. This is only an issue because `mlir_add_dependency` also calls `add_mlir_library_install`. In fact all other uses of `llvm_add_dependency` inside of mlir make sure to explicitly add the install library ([example](https://github.com/llvm/llvm-project/blob/ddb67e68474092435634dd8a8e1c166f6ecd0e1a/mlir/lib/Tools/mlir-tblgen/CMakeLists.txt#L4)) except for `MLIRPDLLParser`.

tl;dr The dependencies of `mlir-pdll` are already being built by the current configuration. The issue is that `MLIRPDLLParser` is not setup to be installed the llvm repo.


Is it possible to install manually it during the CI build?

Reply:
Also, I added `add_mlir_library_install` to the llvm cmakelists to test if that would fix it, and it built the library and linked properly. However, that's probably not a viable solution

Reply:
That sounds like something to fix upstream somehow :/

> Is it possible to install manually it during the CI build?

Maybe? The byo_llvm path is already precarious, so

1. Developers may have to follow some extra steps carefully
2. If we (and upstream LLVM/MLIR) can make that easier, we should

Reply:
Another option is to disable that part of PDLL support in the byo_llvm case. I don't know off the top of my head how we'd do that though.

Reply:
i'm out today; to unblock this and for the sake of incrementalism, i would suggest taking the easy way out for now here (disable PDLL in byo-llvm build) while filing an issue about doing The Right Thing and referencing it in a TODO comment for now.